src/rust/aggregator/src/pipelines/order_history_pipelines.rs

```diff
@@ -209,6 +214,16 @@ impl OrderHistoryPipelines {
             .await
             .map_err(to_pipeline_error)?;
 
+            // Update the batch size based on how many events the current batch contained.
+            let n_events = places.len() + fills.len() + changes.len() + cancels.len();
+            if n_events > TARGET_EVENTS && self.batch_size > BigDecimal::from(1) {
+                self.batch_size = &self.batch_size / BigDecimal::from(10);
+                tracing::debug!("Batch size reduced to {}", self.batch_size);
+            } else if n_events < TARGET_EVENTS / 10 && self.batch_size < BigDecimal::from(MAX_BATCH_SIZE) {
+                self.batch_size = &self.batch_size * BigDecimal::from(10);
```

Comment:
- [ ] Fix all inlined constants to public constants that are used across modules
- [ ] Use 2x on the increase up to saturation, and 1/10th on the back off from saturation